### PR TITLE
added span-link normalization and fixed errors in getTransaction

### PR DESCRIPTION
**Pull request by bryce-b**

### CODE REVIEW

File: x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings.ts [hunk 1]
```diff
@@ -316,20 +315,17 @@ export const spanLinksDetailsMapping = (fields: Partial<Record<string, unknown[]
...
-            id: normalizeValue<string>(fields[SPAN_LINKS_TRACE_ID]),
-          },
-          span: {
-            id: normalizeValue<string>(fields[SPAN_LINKS_SPAN_ID]),
-          },
-        },
-      ],
-    },
-  };
+  return (fields[SPAN_LINKS_TRACE_ID] as string[]).map((v, index) => {
```
Comment (bryce-b) [inline x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings.ts:295]: not sure if this works for OTel

File: x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings.ts [hunk 2]
```diff
@@ -369,7 +365,8 @@ export const transactionMapping = (fields: Partial<Record<string, unknown[]>>) =
       outcome: normalizeValue<EventOutcome>(fields[EVENT_OUTCOME]),
     },
     processor: {
-      event: normalizeValue<string>(fields[PROCESSOR_EVENT]),
+      event: normalizeValue<'transaction'>(fields[PROCESSOR_EVENT]),
+      name: normalizeValue<'transaction'>(fields[PROCESSOR_EVENT]),
```
Comment (miloszmarcinkowski) [inline x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings.ts]:
```suggestion
      name: normalizeValue<'transaction'>(fields[PROCESOR_NAME]),
```

File: x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings.ts (hunk 1 above)
Comment (miloszmarcinkowski) [reply x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings.ts:295]: Thanks for scaffolding this. I don't see a way to test it against real data so far.

**Review by miloszmarcinkowski — COMMENTED** [review]
Looks good to me. Could you please rebase your branch onto mine once I fetch changes from Jenny's branch?

**Comment by miloszmarcinkowski** [comment]
I rebased my branch onto Jenny's. I see that it generated more conflicts in this PR. Let me know if you need any help from my side.

**Review by miloszmarcinkowski — APPROVED** [review]

File: x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings.ts [hunk 3]
```diff
@@ -573,8 +575,27 @@ export const serviceMetadataIconsMapping = (
         name: normalizeValue<string>(fields[CLOUD_PROVIDER]),
       },
     },
-    container: normalizeValue<Container>(fields[CONTAINER]),
-    kubernetes: normalizeValue<Kubernetes>(fields[KUBERNETES]),
+    container: {
```
Comment (bryce-b) [inline x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings.ts]: @miloszmarcinkowski  
There's an issue that needs to be resolved here. The icons only check if `kubernetes:` is not null, so this structure will result in the kubernetes Icon always being shown. i.e. if there are no kubernetes related fields this structure will show up as 
```
kubernetes: {
  pod: {
   name: undefined,
   id: undefined,
   }, 
   ....
}
``` 

What would the best pattern in javascript be to conditionally add each field? something like:
```
 kubernetes: {
      pod: {
       normalizeValue<string>(fields[KUBERNETES_POD_NAME]) ? name: normalizeValue<string>(fields[KUBERNETES_POD_NAME]) : null,
...
      },
```

We may need to reconsider other objects build in this file due to this issue.

Comment (miloszmarcinkowski) [reply x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings.ts]: Good point. The trade-off is that we need to select one property that will define existence for entire object. For given:
```
{
  'kubernetes.pod.id': ['id'],
}
```
this code
```
kubernetes: {
  ...(normalizeValue<string>(fields[KUBERNETES_POD_NAME])
    ? {
        pod: {
          name: normalizeValue<string>(fields[KUBERNETES_POD_NAME]),
          uid: normalizeValue<string>(fields[KUBERNETES_POD_UID]),
        },
      }
    : undefined),
},
```
will produce:
```
{
  kubernetes: undefined
}
```

The effect is that we lose `kubernetes.pod.id`. Maybe instead we should change condition and rely on `kubernetes.pod.name`?

@jennypavlova what do you think?

**Comment by bryce-b** [comment]
@miloszmarcinkowski  I went through and compared these changes to edge-lite and verified all the service meta data icons and details are looking good. The one thing that doesn't look right is the kubernetes namespaces value. I'm not able to figure out where that should be fed through to the `container_details` element. It almost looks like it was removed in some other PR that hasn't made it to edge-lite yet? 
### Edge-lite
Screenshot 2024-09-27 at 14 14 09

### PR 
Screenshot 2024-09-27 at 14 14 13

**Comment by miloszmarcinkowski** [comment]
@bryce-b on OTel sync we decided to move forward with merging PRs as this starts to keep us out of sync. I'll merge it and create a separate comment in my PR with things we discussed here. If you had anything pending, please let me know.

File: x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings.ts (hunk 3 above)
Comment (miloszmarcinkowski) [reply x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings.ts]: @bryce-b thank you for solving it

**Review by miloszmarcinkowski — APPROVED** [review]

**Pull request merged**